### PR TITLE
Chapter 4.1.5: fix incorrect dimensions of tensors in hybrid quantum-classical neural network 

### PR DESCRIPTION
**Pull request by cyx617**
# Changes made

Modified model architecture of the hybrid quantum-classical neural network, as pointed out in [#659](https://github.com/qiskit-community/qiskit-textbook/issues/659)

# Justification

In the original model, the dimensions of the input tensors for fc1 and fc2 layers are inconsistent with the batch size set for model training, which results in long training time and fluctuating loss curve. After the model is updated, it can achieve a better performance in terms of both train/test loss and running time.

### CODE REVIEW

**Review by frankharkins — APPROVED** [review]

**Comment by frankharkins** [comment]
Thank you! For the record, @amyami187 has approved this (I couldn't seem to request the review through github).
